src/api/firebase.js

```diff
@@ -213,12 +213,42 @@ export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 	}
 }
 
-export async function updateItem(listPath, itemId) {
+export async function updateItem(
+	listPath,
+	itemId,
+	dateLastPurchased,
+	totalPurchases,
+	dateNextPurchased,
+	dateCreated,
+) {
+	const dateNextPurchasedJS = dateNextPurchased.toDate();
+	const dateLastPurchasedJS = dateLastPurchased?.toDate();
```

Comment:
Quick questions here: Why are these names with JS at the end? Also, what are the consequences of this line in particular being nullable?

Reply:
We added JS to the end of the names to help identify Firestore timestamps that have been converted to JavaScript dates with the toDate() function, which were later used to update or calculate other time related variables. 
As for line 225, we realized that new items added to the list have a 'null' value in the 'dateLastPurchased' field, and without using optional chaining, the component threw an error.